torch/nn/parallel/distributed.py

```diff
@@ -864,6 +866,44 @@ def __init__(
 
         self._lazy_init_ran = False
 
+        # Register the AccumulaGrad post hooks even if DDP is not compiled. This
+        # can avoid compiling the hooks twice. The Python hooks will be
+        # deregistered later if DDP is not compiled.
+        self._accum_grad_hooks: List[RemovableHandle] = []
+        self._ddp_python_hook = torch._dynamo.config.ddp_python_hook
+        if self._ddp_python_hook:
+            torch._dynamo.config.optimize_ddp = False
```

Comment:
i'm not sure we want to mutate the dynamo config from runtime.

instead, could we just error if this is set?  And expect users to either enable optimize_ddp or compile_ddp?

(its a bigger question of how we want to roll out this feature)

Reply:
+1 I'll vote for having both `optimize_ddp` and `compile_ddp` config flags for now, and sunset both flags when compile DDP is ready for general rollout.

Reply:
Change the behavior to raise an exception.